include/deal.II/base/patterns_tools.h

```diff
@@ -0,0 +1,682 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__patterns_tools_h
+#define dealii__patterns_tools_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/subscriptor.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/std_cxx14/memory.h>
+
+#include <boost/core/demangle.hpp>
```

Comment:
Yes, very nicely. :) 

With STL containers it is a little verbose... But it's useful to debug what goes wrong...